### PR TITLE
Make tracy optional

### PR DESCRIPTION
**Pull request by TomasVotruba**
Ref #218

### CODE REVIEW

File: src/Kdyby/Doctrine/DI/OrmExtension.php [hunk 1]
```diff
@@ -530,9 +533,11 @@ protected function processConnection($name, array $defaults, $isDefault = FALSE)
 		}
 
 		// tracy panel
-		$builder->addDefinition($this->prefix($name . '.diagnosticsPanel'))
-			->setClass('Kdyby\Doctrine\Diagnostics\Panel')
-			->setAutowired(FALSE);
+		if (interface_exists('Tracy\IBarPanel')) {
+			$builder->addDefinition($this->prefix($name . '.diagnosticsPanel'))
+					->setClass('Kdyby\Doctrine\Diagnostics\Panel')
+					->setAutowired(FALSE);
```
Comment (enumag) [inline src/Kdyby/Doctrine/DI/OrmExtension.php]: wrong indentation

File: .travis.yml [hunk 2]
```diff
@@ -1,5 +1,7 @@
 language: php
 
+sudo: false
+
```
Comment (enumag) [inline .travis.yml]: ?

File: src/Kdyby/Doctrine/DI/OrmExtension.php (hunk 1 above)
Comment (TomasVotruba) [inline src/Kdyby/Doctrine/DI/OrmExtension.php]: Thx

File: .travis.yml (hunk 2 above)
Comment (TomasVotruba) [inline .travis.yml]: See commit name

Comment (enumag) [inline .travis.yml]: Yeah, I noticed in the meantime.

**Comment by enumag** [comment]
:+1:

File: src/Kdyby/Doctrine/DI/OrmExtension.php [hunk 3]
```diff
@@ -762,18 +771,19 @@ protected function getServiceBoundManagers(Nette\DI\ServiceDefinition $def)
...
-
-		if (property_exists('Tracy\BlueScreen', 'collapsePaths')) {
-			$blueScreen = 'Tracy\Debugger::getBlueScreen()';
-			$commonDirname = dirname(Nette\Reflection\ClassType::from('Doctrine\Common\Version')->getFileName());
-
-			$init->addBody($blueScreen . '->collapsePaths[] = ?;', array(dirname(Nette\Reflection\ClassType::from('Kdyby\Doctrine\Exception')->getFileName())));
-			$init->addBody($blueScreen . '->collapsePaths[] = ?;', array(dirname(dirname(dirname(dirname($commonDirname)))))); // this should be vendor/doctrine
-			foreach ($this->proxyAutoloaders as $dir) {
-				$init->addBody($blueScreen . '->collapsePaths[] = ?;', array($dir));
+		if (interface_exists('Tracy\IBarPanel')) {
```
Comment (Majkl578) [inline src/Kdyby/Doctrine/DI/OrmExtension.php]: This is ugly. I'd suggest moving it to private method and call it conditionally (or always and return conditionally).

Comment (TomasVotruba) [inline src/Kdyby/Doctrine/DI/OrmExtension.php]: Do you mean sth like?

```php
private function isTracyPresent()
{
    return interface_exists('Tracy\IBarPanel');
}
```

Comment (Majkl578) [inline src/Kdyby/Doctrine/DI/OrmExtension.php]: No, I meant the code inside the condition.
But that method makes sense too (maybe static though). :)

File: src/Kdyby/Doctrine/DI/OrmExtension.php [hunk 4]
```diff
@@ -400,19 +401,22 @@ protected function processEntityManager($name, array $defaults)
...
 				$this->prefix('@' . $name . '.ormConfiguration'),
 				$this->prefix('@' . $name . '.evm'),
 			))
-			->addSetup('?->bindEntityManager(?)', array($this->prefix('@' . $name . '.diagnosticsPanel'), '@self'))
 			->addTag(self::TAG_ENTITY_MANAGER)
 			->addTag('kdyby.doctrine.entityManager')
 			->setAutowired($isDefault)
 			->setInject(FALSE);
 
+		if (interface_exists('Tracy\IBarPanel')) {
```
Comment (fprochazka) [inline src/Kdyby/Doctrine/DI/OrmExtension.php]: you've missed this spot

File: src/Kdyby/Doctrine/DI/OrmExtension.php [hunk 5]
```diff
@@ -762,18 +772,12 @@ protected function getServiceBoundManagers(Nette\DI\ServiceDefinition $def)
...
 
-		if (property_exists('Tracy\BlueScreen', 'collapsePaths')) {
-			$blueScreen = 'Tracy\Debugger::getBlueScreen()';
-			$commonDirname = dirname(Nette\Reflection\ClassType::from('Doctrine\Common\Version')->getFileName());
-
-			$init->addBody($blueScreen . '->collapsePaths[] = ?;', array(dirname(Nette\Reflection\ClassType::from('Kdyby\Doctrine\Exception')->getFileName())));
-			$init->addBody($blueScreen . '->collapsePaths[] = ?;', array(dirname(dirname(dirname(dirname($commonDirname)))))); // this should be vendor/doctrine
-			foreach ($this->proxyAutoloaders as $dir) {
-				$init->addBody($blueScreen . '->collapsePaths[] = ?;', array($dir));
+			if (property_exists('Tracy\BlueScreen', 'collapsePaths')) {
```
Comment (fprochazka) [inline src/Kdyby/Doctrine/DI/OrmExtension.php]: I see no point in checking if `BlueScreen` is present, if `IBarPanel` is already checked.

**Comment by fprochazka** [comment]
Thank you, merged https://github.com/Kdyby/Doctrine/commit/d23ef4d6e236f1d148443f53411e9537d213b3ae

**Comment by TomasVotruba** [comment]
Thanks

**Pull request closed without merging**